src/libsync/syncengine.cpp

```diff
@@ -1301,33 +1306,14 @@ void SyncEngine::restoreOldFiles()
     }
 }
 
-bool SyncEngine::estimateState(QString fn, csync_ftw_type_e t, SyncFileStatus* s)
+SyncFileItem* SyncEngine::findSyncItem(const QString &fileName) const
 {
-    Q_UNUSED(t);
-    QString pat(fn);
-    if( t == CSYNC_FTW_TYPE_DIR && ! fn.endsWith(QLatin1Char('/'))) {
-        pat.append(QLatin1Char('/'));
-    }
-
     Q_FOREACH(const SyncFileItemPtr &item, _syncedItems) {
```

Comment:
i fear this is slow..  linear search  on this function that might be called often.
Is the _syncedItems vector not sorted by filename so a binary search is in order?